contracts/protocol/clearinghouse/ClearingHouse.sol

```diff
@@ -126,35 +133,40 @@ contract ClearingHouse is
         );
     }
 
-    function pause(uint32[] calldata allPoolIds) external onlyGovernanceOrTeamMultisig {
-        _pause();
+    function pause(uint256 numberOfPoolsToUpdateInThisTx) external onlyGovernanceOrTeamMultisig whenNotPaused {
+        if (unpauseLoop.isInProgress()) revert CannotPauseIfUnpauseInProgress();
 
-        // update funding state for all the pools, so that funding payment upto pause moment is recorded
-        for (uint256 i; i < allPoolIds.length; i++) {
-            uint32 poolId = allPoolIds[i];
-            protocol.pools[poolId].vPoolWrapper.updateGlobalFundingState({ useZeroFundingRate: false });
-        }
+        bool completed = pauseLoop.iterate({
+            startAt: 0,
+            endBefore: protocol.poolIds.length,
+            batchSize: numberOfPoolsToUpdateInThisTx,
+            execute: _forEachPoolOnPause
+        });
+
+        if (completed) _pause();
     }
 
-    function unpause(uint32[] calldata allPoolIds) external onlyGovernanceOrTeamMultisig {
-        _unpause();
+    function unpause(uint256 numberOfPoolsToUpdateInThisTx) external onlyGovernanceOrTeamMultisig whenPaused {
+        if (pauseLoop.isInProgress()) revert CannotUnpauseIfPauseInProgress();
 
-        // update funding state for all the pools
-        for (uint256 i; i < allPoolIds.length; i++) {
-            // record the funding payment as zero for the entire duration for which clearing house was paused.
-            protocol.pools[allPoolIds[i]].vPoolWrapper.updateGlobalFundingState({ useZeroFundingRate: true });
-        }
+        bool completed = unpauseLoop.iterate({
+            startAt: 0,
+            endBefore: protocol.poolIds.length,
+            batchSize: numberOfPoolsToUpdateInThisTx,
+            execute: _forEachPoolOnUnpause
+        });
+
+        if (completed) _unpause();
     }
 
     /// @inheritdoc IClearingHouseOwnerActions
-    function withdrawProtocolFee(address[] calldata wrapperAddresses) external {
-        uint256 totalProtocolFee;
-        for (uint256 i = 0; i < wrapperAddresses.length; i++) {
-            uint256 wrapperFee = IVPoolWrapper(wrapperAddresses[i]).collectAccruedProtocolFee();
-            emit Account.ProtocolFeesWithdrawn(wrapperAddresses[i], wrapperFee);
-            totalProtocolFee += wrapperFee;
-        }
-        protocol.settlementToken.safeTransfer(teamMultisig(), totalProtocolFee);
+    function withdrawProtocolFee(uint256 numberOfPoolsToUpdateInThisTx) external {
```

Comment:
Should we add iterations for withdrawProtocolFee? We lose the functionality being able to withdraw fees for specific pools.